cpp/src/arrow/compute/exec/aggregate_node.cc

```diff
@@ -326,46 +446,86 @@ class ScalarAggregateNode : public ExecNode, public TracedNode {
   }
 
  private:
-  Status Finish() {
-    auto scope = TraceFinish();
+  Status ReconstructAggregates() {
+    const auto& input_schema = *inputs()[0]->output_schema();
+    auto exec_ctx = plan()->query_context()->exec_context();
+    for (size_t i = 0; i < kernels_.size(); ++i) {
+      std::vector<TypeHolder> in_types;
```

Comment:
Do we not have this variable saved somewhere? Since this wouldn't change between segments it feels weird to recompute this every time we reset the state

Reply:
Fixed.